src/ui/public/visualize/loader/pipeline_helpers/has_minimal_columns.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Vis } from 'ui/vis';
+
+// Vis types that have a `showMetricsAtAllLevels` param (e.g. data table) should tell
+// the build pipeline whether to add metrics for each bucket to the dimensions
+// based on the param value. Vis types without this param should default to returning
+// all columns if they are hierarchical.
+export function hasMinimalColumns(vis: Vis): boolean {
```

Comment:
as his is only affecting table .... 
doent checking for `vis.params.showMetricsAtAllLevels`do the trick ?